packages/components/src/Combobox/Combobox.types.ts

```diff
@@ -135,6 +145,11 @@ export interface ComboboxOptionProps {
       isSelected: boolean;
     },
   ) => React.ReactNode;
+
+  /**
+   * Callback function invoked when the option is clicked.
+   */
+  readonly onOptionClick?: () => void;
```

Comment:
sorry ignore my other comment. I was reading the diff wrong and though this was somewhere else!

my remaining feedback is that `onOptionClick` feels redundant to me. we're going to have

```
<Combobox ...>
  <Combobox.Option onOptionClick={() => ...} />
</Combobox>
```

I'd argue it's obvious that it belongs to the `Option`

`onClick` should suffice.

Reply:
one other thing, we might as well pass the clicked item as an arg. there's no downside to getting that extra information. if you don't want to use it, just ignore it.